broker/src/test/java/io/zeebe/broker/system/SystemContextTest.java

```diff
@@ -131,6 +131,20 @@ public void shouldNotThrowExceptionIfSnapshotPeriodIsEqualToOneMinute() {
         .isEqualTo(Duration.ofMinutes(1));
   }
 
+  @Test
+  public void shouldInvalidateConfigIfUseMmapTrueWithReplication() {
+    // given
+    final BrokerCfg brokerCfg = new BrokerCfg();
+    brokerCfg.getData().setUseMmap(true);
+    brokerCfg.getCluster().setClusterSize(2);
+    brokerCfg.getCluster().setReplicationFactor(2);
+
+    // then
+    expectedException.expect(IllegalStateException.class);
```

Comment:
Personally, not a fan of the expectedException rule. I would rather use `Assertions.assertThatThronwBy(..)` as it makes clearer where the exception is expected. 
Fee free to ignore this comment, because it fits to the other test methods in this class.

Reply:
Hm, I also prefer assert but in this case I kept it as you said, since it was consistent with the rest of the tests :sweat_smile: I'd also lean towards keeping things consistent over my personal preference